tests/metrics/functional/test_image_gradients.py

```diff
@@ -0,0 +1,253 @@
+import torch
+import pytest
+import torch.nn.functional as F
+
+from pytorch_lightning.metrics.functional.image_gradients import image_gradients
+
+
+def test_invalid_input_img_type():
+    """Test Whether the module successfully handles invalid input data type"""
+    invalid_dummy_input = [[1, 2, 3], [4, 5, 6], [7, 8, 9]]
+    device = torch.device("cpu")
+    with pytest.raises(TypeError):
+        image_gradients(invalid_dummy_input, device)
+
+
+def test_invalid_input_device_type():
+    """Test wheather the module successfully handles invald input device type"""
+    BATCH_SIZE = 1
+    HEIGHT = 5
+    WIDTH = 5
+    CHANNELS = 1
+
+    device = "cpu"
+
+    image = torch.arange(0, BATCH_SIZE * HEIGHT * WIDTH * CHANNELS, dtype=torch.float32, device=device)
+    image = torch.reshape(image, (HEIGHT, WIDTH))
+
+    with pytest.raises(TypeError):
+        image_gradients(image, device)
+
+
+def test_invalid_input_ndims():
+    """
+    Test whether the module successfully handles invalid number of dimensions
+    of input tensor
+    """
+
+    BATCH_SIZE = 1
+    HEIGHT = 5
+    WIDTH = 5
+    CHANNELS = 1
+
+    device = torch.device("cpu")
+
+    image = torch.arange(0, BATCH_SIZE * HEIGHT * WIDTH * CHANNELS, dtype=torch.float32, device=device)
+    image = torch.reshape(image, (HEIGHT, WIDTH))
+
+    with pytest.raises(RuntimeError):
+        image_gradients(image, device)
+
+
+def test_multi_batch_image_gradients():
+    """Test whether the module correctly calculates gradients for known input
+    with non-unity batch size.Example input-output pair taken from TF's implementation of i
+    mage-gradients
+    """
+
+    BATCH_SIZE = 5
+    HEIGHT = 5
+    WIDTH = 5
+    CHANNELS = 1
+
+    device = torch.device("cpu")
+
+    single_channel_img = torch.arange(0, 1 * HEIGHT * WIDTH * CHANNELS, dtype=torch.float32, device=device)
+    single_channel_img = torch.reshape(single_channel_img, (CHANNELS, HEIGHT, WIDTH))
+    image = torch.stack([single_channel_img for _ in range(BATCH_SIZE)], dim=0)
+
+    true_dy = [
```

Comment:
Could you use true_dy = [[....],
                                          [.....]] instead. It is easier to read.

Reply:
Sure. Will do that.

Reply:
made the recommended changes in the latest commit